src/Analyzers/CSharp/CodeFixes/UseCollectionExpression/CSharpUseCollectionExpressionForBuilderCodeFixProvider.cs

```diff
@@ -82,6 +83,15 @@ internal partial class CSharpUseCollectionExpressionForBuilderCodeFixProvider()
         foreach (var (statement, _) in analysisResult.Matches)
             subEditor.RemoveNode(statement);
 
+        //Add the leading trivia from the declaration of the builder to the invocation where we convert the builder to a collection.
+        var dummyObjectCreationDeclaration = dummyObjectCreation.Ancestors().Take(3);
```

Comment:
Why `Take(3)`?At least doc what that number means or turn it into a constant. But I suspect, that this logic is a bit too specific to a single line comment case. What if instead of
```cs
// Leading
var builder = ImmutableArray.CreateBuilder<Type>();
```
you have
```cs
/* Leading */ var builder = ImmutableArray.CreateBuilder<int>();
```
Yeah, Cyrus said in the issue that this is not a common patterns, but a) it mught crash that logic if there are less than 3 trivia elements and b) I don't think is is complicated to handle this case. Your trivia in single line comment case looks like this:
```cs
    // Comment
    <- white spaces here
```
What you need to do is remove that white space to preserve correct formatting after fix. I think, there is a `WithoutTrailingWhiteSpace()` extension method for syntax trivia or similar

Reply:
I've added a const, and clarified in the preceding comment what it does. It however didn't have to do with taking trivia but taking statements before a `new()` declaration. So a complete statement would be `ImmutableArray<int> i = new()`
taking 3 ancestors of the new() declaration would then result in `ImmutableArray<int> i =`